### ISSUE TITLE
ansible-pull does not evaluate inventory

### ISSUE THREAD

**Issue by opoplawski** [ansible/ansible#32455]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ansible-pull

##### ANSIBLE VERSION

```
ansible 2.4.1.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug  2 2017, 11:05:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### OS / ENVIRONMENT
Scientific Linux 7.4

##### SUMMARY
We've been running ansible-pull with;
````
/usr/bin/ansible-pull --url ssh://git@XXX --key-file XXX -C master -i inventory --extra-vars=ansible_pull=1
````

Which has been working fine until 2.4.  Now the inventory is not processed.  If I patch ansible/inventory/manager.py to dump the inventory.hosts in _evaluate_patterns(), hosts consists of just 'localhost'.

strace shows:
````
chdir("/root/.ansible/pull/barry.cora.nwra.com") = 0
stat("/root/inventory", 0x7fffb2910760) = -1 ENOENT (No such file or directory)
````

So looks like the path to the inventory file gets fixed too soon?

##### STEPS TO REPRODUCE

```yaml

```

##### EXPECTED RESULTS
See inventory from checked out repository.

##### ACTUAL RESULTS

```
[WARNING]: get_hosts pattern=all
 [WARNING]: _evaluate_patterns patterns=['all']
 [WARNING]: _evaluate_patterns p=all
 [WARNING]: _evaluate_patterns hosts={u'localhost': localhost}
 [WARNING]: _match_one_pattern pattern=all, expr=all, slice=None
 [WARNING]: _evaluate_patterns patterns=[u'localhost', u'barry', u'barry.cora.nwra.com',
u'127.0.0.1']
 [WARNING]: _evaluate_patterns p=localhost
 [WARNING]: _evaluate_patterns p=barry
 [WARNING]: _match_one_pattern pattern=barry, expr=barry, slice=None
 [WARNING]: Could not match supplied host pattern, ignoring: barry
 [WARNING]: _evaluate_patterns p=barry.cora.nwra.com
 [WARNING]: _match_one_pattern pattern=barry.cora.nwra.com, expr=barry.cora.nwra.com, slice=None
 [WARNING]: Could not match supplied host pattern, ignoring: barry.cora.nwra.com
```

**Comment by opoplawski** [comment]
Hmm, this does seem to be fixed in current devel branch.

**Comment by opoplawski** [comment]
Fixed in 2.4.2.0 as well.  Closing.

**Comment by bcoca** [comment]
fixed by #32135
